### ISSUE TITLE
last.ckpt

### ISSUE THREAD

**Issue by chmxu** [ZhangYuanhan-AI/visual_prompt_retrieval#8]
Thank you for this inspiring work! I want to know which last.ckpt weight for VQGAN should be used in the Step3 of supervised prompt retrieval?

**Comment by ZhangYuanhan-AI** [comment]
Hi, thanks for your interests!

Please download pretrained VQGAN codebook checkpoint and config [_vqgan_imagenet_f16_1024_](https://heibox.uni-heidelberg.de/d/8088892a516d4e3baf92/?p=%2F), place both _last.ckpt_ and _model.yaml_ on the repository root. 

And the pre-trained model we use is this version: 
<img width="545" alt="image" src="https://github.com/ZhangYuanhan-AI/visual_prompt_retrieval/assets/18485270/9d24ef0c-cf80-4b81-803d-e3f91d45c41a">

**Comment by chmxu** [comment]
Thank you!

**Comment by chmxu** [comment]
Hi sorry to bother you again. When conducting Step3 evaluation I find that each sample has only one grid sample. In evaluate_semgnetation.py each canvas fetched in line128 has 1 element in the 0-th dim.

I assume each sample should be composed with 50 other most similar samples for the input grid of inpainting network, so that the performance can be sorted and used for following steps. I am not sure if I have any misunderstanding. Can you help me with this problem? Thank you!

**Comment by ZhangYuanhan-AI** [comment]
> evaluate_semgnetation.py

1. What do you mean "one grid sample"
2. "evaluate_semgnetation.py" is from which project?

**Comment by chmxu** [comment]
Thank you for your reply. The problem is that when I conduct Step4 to get positive and negative samples, I find that the ```iou_dict``` is empty after deleting similarity of each sample itself (line41-line44 in ```get_positive_negative.py```). So I suspect that there are some problems when running ```./evlaluate/evaluate_segmentation.py``` in Step3, but I'm not very sure about that.

**Comment by chmxu** [comment]
Moreover, [L141 in ./evlaluate/evaluate_segmentation.py](https://github.com/ZhangYuanhan-AI/visual_prompt_retrieval/blob/8c307493d4ac8f2b5652345b9004b21703eb9294/evaluate/evaluate_segmentation.py#L141) should generate at most 3 images for each idx. However I only get visualization with grid_idx 0 for each idx.

**Comment by ZhangYuanhan-AI** [comment]
https://github.com/ZhangYuanhan-AI/visual_prompt_retrieval/blob/8c307493d4ac8f2b5652345b9004b21703eb9294/evaluate/pascal_dataloader.py#L100

You can get more grid by changing the value here

**Comment by chmxu** [comment]
Is changing it to range(50) appropriate, or is this same to your original setting?

**Comment by ZhangYuanhan-AI** [comment]
it should be one, if your in-context example is one image-label pair.

**Comment by chmxu** [comment]
Yes, but if it's one then by deleting self-similarity the iou_dict for each sample_idx is empty. In this way how to select pos/neg samples for training?

**Comment by chmxu** [comment]
<img width="1241" alt="image" src="https://github.com/ZhangYuanhan-AI/visual_prompt_retrieval/assets/17843174/c9881274-b776-4070-b93d-362386772fb8">
This is what I get when using the original code. Not sure if there is any problems.

**Comment by ZhangYuanhan-AI** [comment]
can you check the "XXXtop50-similarity.json ?"

**Comment by chmxu** [comment]
<img width="1009" alt="image" src="https://github.com/ZhangYuanhan-AI/visual_prompt_retrieval/assets/17843174/141e8d38-df99-4d2e-870a-e133c3778da7">

Sure. It seems have no problem.

**Comment by ZhangYuanhan-AI** [comment]
So can you debug this part?

<img width="703" alt="image" src="https://github.com/ZhangYuanhan-AI/visual_prompt_retrieval/assets/18485270/0145a7ee-2f80-48bb-8aa0-79c96b619122">

**Comment by chmxu** [comment]
<img width="1300" alt="image" src="https://github.com/ZhangYuanhan-AI/visual_prompt_retrieval/assets/17843174/885c501a-7faa-4a4c-b921-202c1cce49b1">

Can you please help me check if this ```log.txt```, which is loaded in ```get_positive_negative.py``` as variable ```metas```, is proper?

**Comment by ZhangYuanhan-AI** [comment]
Looks fine

**Comment by chmxu** [comment]
I've uploaded a full log.txt here https://drive.google.com/file/d/19Nq9QBqN2D0clF-0cvfneSn65aKcowXy/view?usp=sharing. If this data is fine then I have no clue why I can't get reasonable result from Step4.

**Comment by ZhangYuanhan-AI** [comment]
> Yes, but if it's one then by deleting self-similarity the iou_dict for each sample_idx is empty. In this way how to select pos/neg samples for training?

Any thoughts about why does iou_dict become empty?

**Comment by chmxu** [comment]
I'll look into the code later. Hope I can solve this problem with your help.

**Comment by ZhangYuanhan-AI** [comment]
Ok, I understand why.

In the repository located at [https://github.com/ZhangYuanhan-AI/visual_prompt_retrieval/blob/seg_val2train/SupPR.md](https://github.com/ZhangYuanhan-AI/visual_prompt_retrieval/blob/seg_val2train/SupPR.md), there are two 'evaluation stages.' 

![Evaluation Stage](https://github.com/ZhangYuanhan-AI/visual_prompt_retrieval/assets/18485270/97fe733a-f9ba-4499-856a-607a6c4ca7af)

During the first 'evaluation' stage, our goal is to obtain the Intersection over Union (IoU) related to the top 50 similar examples based on cosine similarity. Therefore, in the script at [https://github.com/ZhangYuanhan-AI/visual_prompt_retrieval/blob/8c307493d4ac8f2b5652345b9004b21703eb9294/evaluate/pascal_dataloader.py#L100](https://github.com/ZhangYuanhan-AI/visual_prompt_retrieval/blob/8c307493d4ac8f2b5652345b9004b21703eb9294/evaluate/pascal_dataloader.py#L100), we should set the number to 50.

In the second 'evaluation' stage, the aim is to obtain the IoU of the top 1 similar example, which utilizes a trained feature extractor. For this stage, the number should be set to 1.

Therefore, please ensure that the code is correctly adjusted according to the stage you are in.

**Comment by chmxu** [comment]
Thank you!
